### PR TITLE
OpState + RVC OpState - add tests for new pause and resume compatible states

### LINKED ISSUE #31393: Update the OperationalState and RvcOperationalState test scripts to test for new pause and resume compatible states
Update the OperationalState and RvcOperationalState cluster test scripts according to test-plan changes in [this PR](https://github.com/CHIP-Specifications/chip-test-plans/pull/3750). That PR allows certain states to be pause and resume compatible.

In summary 
- Update the cluster revisions.
- Update the test scripts.
- Add the GoHome command test script TC-2.4

### PR DESCRIPTION
**Pull request by hicklin**
Fixes #31393.

This PR updates the Operational State and RVC Operational State cluster test scripts based on the test-plan changes in [this PR](https://github.com/CHIP-Specifications/chip-test-plans/pull/3750).

In summary, this PR 
- Removes the RVC Operational state test case 2.2
- Updates the Operational State test case 2.3
- Updates the RVC Operational State test case 2.3
- Adds the GoHome command test script TC-2.4

### CODE REVIEW

File: src/python_testing/TC_OPSTATE_2_3.py [hunk 1]
```diff
@@ -80,74 +80,181 @@ async def test_TC_OPSTATE_2_3(self):
...
+                             "OperationalState(%s) should be the state before pause (%s)" % operational_state, old_opstate_dut)
 
-        self.print_step(12, "Send Resume command")
+        self.print_step(13, "Send Resume command")
         ret = await self.send_resume_cmd()
         asserts.assert_equal(ret.commandResponseState.errorStateID, Clusters.OperationalState.Enums.ErrorStateEnum.kNoError,
                              "errorStateID(%s) should be NoError(0x00)" % ret.commandResponseState.errorStateID)
 
-        self.print_step(13, "Manually put the device in a state where it cannot receive a Pause command")
+        if self.check_pics("OPSTATE.S.M.RESUME_AFTER_ERR"):
```
Comment (plauric) [inline src/python_testing/TC_OPSTATE_2_3.py:143]: This PR makes changes to the wrong file :) 

There's a TC 2.3 test script for the **(base) OpState** cluster which needs small changes - see steps 15-18 in the test plan PR: https://github.com/CHIP-Specifications/chip-test-plans/pull/3750/files

There's also a TC 2.3 script for the **RVC OpState** cluster, where most of the changes below need to go.

Comment (hicklin) [reply src/python_testing/TC_OPSTATE_2_3.py]: I overwrote my previous updates to the opstate tests. I have now fixed this. Please revisit both.

File: src/python_testing/TC_RVCOPSTATE_2_4.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        return "WaterTankEmpty(0x44)"
+    elif error_enum == Clusters.RvcOperationalState.Enums.ErrorStateEnum.kWaterTankMissing:
+        return "WaterTankMissing(0x45)"
+    elif error_enum == Clusters.RvcOperationalState.Enums.ErrorStateEnum.kWaterTankLidOpen:
+        return "WaterTankLidOpen(0x46)"
+    elif error_enum == Clusters.RvcOperationalState.Enums.ErrorStateEnum.kMopCleaningPadMissing:
+        return "MopCleaningPadMissing(0x47)"
+
+
+class TC_RVCOPSTATE_2_3(MatterBaseTest):
```
Comment (plauric) [inline src/python_testing/TC_RVCOPSTATE_2_4.py:76]:
```suggestion
class TC_RVCOPSTATE_2_4(MatterBaseTest):
```

File: src/python_testing/TC_RVCOPSTATE_2_4.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+    @async_test_body
+    async def test_TC_RVCOPSTATE_2_4(self):
+
+        asserts.assert_true('PIXIT_ENDPOINT' in self.matter_test_config.global_test_params,
+                            "PIXIT_ENDPOINT must be included on the command line in "
+                            "the --int-arg flag as PIXIT_ENDPOINT:<endpoint>")
+
+        asserts.assert_true(self.check_pics("RVCOPSTATE.S.A0004"), "RVCOPSTATE.S.A0004 must be supported")
+        asserts.assert_true(self.check_pics("RVCOPSTATE.S.C80.Rsp"), "RVCOPSTATE.S.C80.Rsp must be supported")
```
Comment (plauric) [inline src/python_testing/TC_RVCOPSTATE_2_4.py:122]:
```suggestion
        asserts.assert_true(self.check_pics("RVCOPSTATE.S.C04.Rsp"), "RVCOPSTATE.S.C04.Rsp must be supported")
```

I assume this is the PICS for the GoHome command. BTW, the test plan PR (3750) has a bug re. this, and I'll fix it shortly.

Comment (hicklin) [reply src/python_testing/TC_RVCOPSTATE_2_4.py:122]: Is `C04` meant to match with the [command ID as described in the spec](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/a0686eddba3b04212c3b0f8ec224e8a13ae6dce2/src/app_clusters/OperationalState_RVC.adoc?plain=1#L198)? If so, it should be `C08`. If not, can you explain where the `C04` comes from?

File: src/python_testing/TC_OPSTATE_2_3.py [hunk 4]
```diff
@@ -144,10 +144,22 @@ async def test_TC_OPSTATE_2_3(self):
...
                              "errorStateID(%s) should be CommandInvalidInState(0x03)" % ret.commandResponseState.errorStateID)
 
-        self.print_step(15, "Manually put the device in a state where it cannot receive a Resume command")
+        self.print_step(15, "Send Resume command")
+        ret = await self.send_resume_cmd()
+        asserts.assert_equal(ret.commandResponseState.errorStateID,
+                             Clusters.OperationalState.Enums.ErrorStateEnum.kCommandInvalidInState,
+                             "errorStateID(%s) should be CommandInvalidInState(0x03)" % ret.commandResponseState.errorStateID)
+
+        self.print_step(16, "Manually put the device in a the Error(0x03) operational state")
```
Comment (plauric) [inline src/python_testing/TC_OPSTATE_2_3.py:153]:
```suggestion
        self.print_step(16, "Manually put the device in the Error(0x03) operational state")
```

File: src/python_testing/TC_RVCOPSTATE_2_3.py [hunk 5]
```diff
@@ -80,74 +80,181 @@ async def test_TC_RVCOPSTATE_2_3(self):
...
         operational_state = await self.read_mod_attribute_expect_success(endpoint=self.endpoint,
                                                                          attribute=attributes.OperationalState)
-        logging.info("OperationalState: %s" % (operational_state))
+        logging.info("OperationalState: %s" % operational_state)
         asserts.assert_equal(operational_state, Clusters.OperationalState.Enums.OperationalStateEnum.kPaused,
                              "OperationalState(%s) should be Paused(0x02)" % operational_state)
 
-        if self.check_pics("RVCOPSTATE.S.A0002"):
-            self.print_step(6, "Read CountdownTime attribute")
+        if self.check_pics("OPSTATE.S.A0002"):
```
Comment (plauric) [inline src/python_testing/TC_RVCOPSTATE_2_3.py:100]:
```suggestion
        if self.check_pics("RVCOPSTATE.S.A0002"):
```

File: src/python_testing/TC_RVCOPSTATE_2_3.py [hunk 6]
```diff
@@ -80,74 +80,181 @@ async def test_TC_RVCOPSTATE_2_3(self):
...
-            self.print_step(8, "Read CountdownTime attribute")
+            self.print_step(9, "Read CountdownTime attribute")
             countdown_time = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.CountdownTime)
-            logging.info("CountdownTime: %s" % (countdown_time))
+            logging.info("CountdownTime: %s" % countdown_time)
             asserts.assert_true(countdown_time != 0 or countdown_time == NullValue,
                                 "invalid CountdownTime(%s). Must be a non zero integer, or null" % countdown_time)
-            asserts.assert_equal(countdown_time, initial_countdown_time,
-                                 "CountdownTime(%s) does not equal to the intial CountdownTime (%s)" % (countdown_time, initial_countdown_time))
+            asserts.assert_equal(countdown_time, initial_countdown_time, "CountdownTime(%s) does not equal to the intial CountdownTime(%s)"
```
Comment (plauric) [inline src/python_testing/TC_RVCOPSTATE_2_3.py:118]:
```suggestion
            asserts.assert_equal(countdown_time, initial_countdown_time, "CountdownTime(%s) not equal to the initial CountdownTime(%s)"
```

File: src/python_testing/TC_RVCOPSTATE_2_3.py [hunk 7]
```diff
@@ -80,74 +80,181 @@ async def test_TC_RVCOPSTATE_2_3(self):
...
+                                 "errorStateID(%s) should be NoError(0x00)" % ret.commandResponseState.errorStateID)
+
+            self.print_step(23, "Read OperationalState attribute")
+            operational_state = await self.read_mod_attribute_expect_success(endpoint=self.endpoint,
+                                                                             attribute=attributes.OperationalState)
+            logging.info("OperationalState: %s" % operational_state)
+            asserts.assert_equal(operational_state, Clusters.OperationalState.Enums.OperationalStateEnum.kRunning,
+                                 "OperationalState(%s) should be Running(0x01)" % operational_state)
+
+        self.print_step(24, "Manually put the device in the Stopped(0x00) operational state")
```
Comment (plauric) [inline src/python_testing/TC_RVCOPSTATE_2_3.py:186]: Each of these manual steps, and the testing that follows them, needs to guarded by the respective PICS code.

File: src/python_testing/TC_RVCOPSTATE_2_4.py (hunk 3 above)
Comment (plauric) [reply src/python_testing/TC_RVCOPSTATE_2_4.py:122]: @hicklin Sorry, that was a mistake. I edited my suggestion. The command id is 0x80 so 128 decimal.

File: src/python_testing/TC_RVCOPSTATE_2_3.py [hunk 8]
```diff
@@ -59,13 +137,16 @@ async def test_TC_RVCOPSTATE_2_3(self):
         asserts.assert_true(self.check_pics("RVCOPSTATE.S.A0004"), "RVCOPSTATE.S.A0004 must be supported")
         asserts.assert_true(self.check_pics("RVCOPSTATE.S.C00.Rsp"), "RVCOPSTATE.S.C00.Rsp must be supported")
         asserts.assert_true(self.check_pics("RVCOPSTATE.S.C03.Rsp"), "RVCOPSTATE.S.C03.Rsp must be supported")
+        # This command SHALL be supported by an implementation if any of the other commands are supported (6.5)
+        asserts.assert_true(self.check_pics("OPSTATE.S.C04.Rsp"), "OPSTATE.S.C04.Rsp must be supported")
```
Comment (plauric) [inline src/python_testing/TC_RVCOPSTATE_2_3.py:141]:
```suggestion
        asserts.assert_true(self.check_pics("RVCOPSTATE.S.C04.Rsp"), "RVCOPSTATE.S.C04.Rsp must be supported")
```

File: src/python_testing/TC_RVCOPSTATE_2_4.py [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+
+    @async_test_body
+    async def test_TC_RVCOPSTATE_2_4(self):
+
+        asserts.assert_true('PIXIT_ENDPOINT' in self.matter_test_config.global_test_params,
+                            "PIXIT_ENDPOINT must be included on the command line in "
+                            "the --int-arg flag as PIXIT_ENDPOINT:<endpoint>")
+
+        asserts.assert_true(self.check_pics("RVCOPSTATE.S.A0004"), "RVCOPSTATE.S.A0004 must be supported")
+        asserts.assert_true(self.check_pics("OPSTATE.S.C04.Rsp"), "OPSTATE.S.C04.Rsp must be supported")
```
Comment (plauric) [inline src/python_testing/TC_RVCOPSTATE_2_4.py:122]:
```suggestion
        asserts.assert_true(self.check_pics("RVCOPSTATE.S.C04.Rsp"), "RVCOPSTATE.S.C04.Rsp must be supported")
```

**Review by plauric — APPROVED** [review]

**Review by rbultman — APPROVED** [review]

**Review by cecille — COMMENTED** [review]
Discussed this with Lucus earlier, but all these opstate tests
1) Should be coming from a common function since we have the ability to do this in python
2) Need to be enabled in the CI. Which means all the folks involved in the opstate cluster development need to get together to find a way to trigger the manual pieces if this script in the example apps so they can be enabled in the CI

Right now, QA won't be able to run these.

See https://github.com/project-chip/connectedhomeip/issues/31561

File: src/python_testing/TC_RVCOPSTATE_2_3.py [hunk 10]
```diff
@@ -28,8 +28,60 @@
...
+    elif error_enum == Clusters.RvcOperationalState.Enums.ErrorStateEnum.kWaterTankEmpty:
+        return "WaterTankEmpty(0x44)"
+    elif error_enum == Clusters.RvcOperationalState.Enums.ErrorStateEnum.kWaterTankMissing:
+        return "WaterTankMissing(0x45)"
+    elif error_enum == Clusters.RvcOperationalState.Enums.ErrorStateEnum.kWaterTankLidOpen:
+        return "WaterTankLidOpen(0x46)"
+    elif error_enum == Clusters.RvcOperationalState.Enums.ErrorStateEnum.kMopCleaningPadMissing:
+        return "MopCleaningPadMissing(0x47)"
+
+
```
Comment (cecille) [inline src/python_testing/TC_RVCOPSTATE_2_3.py:78]:
```suggestion
    def pics_TC_RVCOPSTATE_2_3(self) -> list[str]:
        return ["RVCOPSTATE.S"]

```

File: src/python_testing/TC_RVCOPSTATE_2_4.py [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+    elif error_enum == Clusters.RvcOperationalState.Enums.ErrorStateEnum.kWaterTankEmpty:
+        return "WaterTankEmpty(0x44)"
+    elif error_enum == Clusters.RvcOperationalState.Enums.ErrorStateEnum.kWaterTankMissing:
+        return "WaterTankMissing(0x45)"
+    elif error_enum == Clusters.RvcOperationalState.Enums.ErrorStateEnum.kWaterTankLidOpen:
+        return "WaterTankLidOpen(0x46)"
+    elif error_enum == Clusters.RvcOperationalState.Enums.ErrorStateEnum.kMopCleaningPadMissing:
+        return "MopCleaningPadMissing(0x47)"
+
+
```
Comment (cecille) [inline src/python_testing/TC_RVCOPSTATE_2_4.py:76]:
```suggestion
    def pics_TC_RVCOPSTATE_2_4(self) -> list[str]:
        return ["RVCOPSTATE.S"]

```

File: src/python_testing/TC_RVCOPSTATE_2_3.py (hunk 10 above)
Comment (plauric) [reply src/python_testing/TC_RVCOPSTATE_2_3.py:78]: I think the return should also include the expansions of these test plan macros: {PICS_SC_PAUSE}, {PICS_SC_RESUME}, {PICS_SC_OPERATIONALCOMMANDRESPONSE}

File: src/python_testing/TC_RVCOPSTATE_2_4.py (hunk 11 above)
Comment (plauric) [reply src/python_testing/TC_RVCOPSTATE_2_4.py:76]: I think the return should also include the expansions of these test plan macros: {PICS_SC_GOHOME},  {PICS_SC_OPERATIONALCOMMANDRESPONSE}, {PICS_SA_OPERATIONALSTATE}

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**